draft-irtf-cfrg-aead-limits.md

```diff
@@ -344,77 +362,99 @@ v * 2^64 + (2l * (v + q))^2 <= p * 2^128
+very small probability. The success probability in attacking one of these many
+independent user keys can be generically bounded by the success probability of
+attacking a single user multiplied by the number of users present {{MUSecurity}}, {{GCM-MU}}.
+Absent concrete multi-user bounds, this means the attacker advantage in the multi-user
 setting is the product of the single-user advantage and the number of users.
 
 This section summarizes the confidentiality and integrity bounds and limits for
-the same algorithms as in {{limits}}, except in the multi-user setting. The CL
+the same algorithms as in {{su-limits}} for the multi-user setting. The CL
 and IL values bound the total number of encryption and forgery queries (q and v).
 Alongside each value, we also specify these bounds.
 
 ## AEAD_AES_128_GCM and AEAD_AES_256_GCM
 
 Concrete multi-user bounds for AEAD_AES_128_GCM and AEAD_AES_256_GCM exist
-due to {{?GCM-MU=DOI.10.1145/3243734.3243816}}. AES-GCM without nonce
-randomization is also discussed in {{?GCM-MU}}, though this section does not
+due to {{GCM-MU2}}. AES-GCM without nonce
+randomization is also discussed in {{GCM-MU2}}, though this section does not
 include those results as they do not apply to protocols such as TLS 1.3 {{?RFC8446}}.
 
-### Confidentiality Limit
+For this AEAD, n = 128, t = 128, and r = 96; the key length is k = 128 or k = 256.
 
-<!-- From (1) in {{GCM-MU}}, assuming n=2^7, \sigma = (v+q)*l, B = \sigma/u, dropping the last term
-  (with denominator 2^(k+n), and dropping the first term since the adversary's
-  offline work dominates -->
-~~~
-CA <= ((v + q) * l)^2 / (u * 2^128)
-~~~
+### Authenticated Encryption Security Limit
 
-This implies the following limit:
+<!--
+    From {{GCM-MU2}} Theorem 4.3.
+
+    Let:
+        - #blocks encrypted/verified overall:   \sigma = (q + v) * l
+        - worst-case  o (offline work), q+v, \sigma <= 2^95
+          (Theorem 4.3 requires q <= 2^(1-e)r ; this yields e >= 0.0104, hence
+          d = 1,5/e -1 <= 143 <= 2^8.)
 
+    We can simplify as follows:
+        - Note: Last term is 2^-48; hence any other term <= 2^-50 is negligible.
+        - 1st term (../2^k):  roughly <= 2^8 * (o + q+v + \sigma) / 2^k
+           roughly <= (o + (q+v)*l) / 2^(k-8)
+          This is negligible for k = 256.
+          For k = 128, it is negligible if o, (q+v)*l <= 2^70.
+          For o <= 2^70 and B >= 2^8, it is dominated by the 2nd term;
+            we assume that and hence omit the 1st term.
+        - 2nd term (../2^n):  roughly  = \sigma*B/2^127
+        - 3rd term (../2^2n):  <= 2^-160, negligible.
+        - 4th term (../2^(k+n)):  roughly <= (\sigma^2 + 2o(q+v)) / 2^256
+          <= 2^-64, negligible.
+        - 5th term (2^(-r/2)):  = 2^48
+-->
 ~~~
-v + q <= sqrt(p * u * 2^128) / l
+AE <= (q+v)*l*B / 2^127 + 1/2^48
 ~~~
 
-### Integrity Limit
+This implies the following limit (where the adversary attack probability `p` is
+always `>= 2^-48`):
```

Comment:
Do we allow the first term to reach 2^-48, because that means p being 2^-48 in the below results in AE being 2^-47.

Reply:
I wanted to express here that "p is always >= 2^-48", but Chris rightfully pointed out that deriving the limit based on both terms (.../2^127 _and_ 1/2^48) is much cleaner. Changed that in 5d3cc7b.